### PR TITLE
Feat/cog

### PR DESCRIPTION
**Pull request by wildintellect**
**Related Issue(s):**
#2 and #10 

**Description:**
Added functions to ungzip original data and convert raster files to cloud optimized geotiff (COG) with gdal_translate.

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

File: src/azure/.gitignore [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+bin
+obj
+appsettings.json
+local.settings.json
+
+# Azurite artifacts
+__blobstorage__
+__queuestorage__
+__azurite_db*__.json
+.python_packages
```
Comment (tracetechnical) [inline src/azure/.gitignore:135]: minor: newline

File: src/stactools/palsar/stac.py [hunk 2]
```diff
@@ -17,42 +23,54 @@ def create_collection() -> Collection:
...
-            roles=[
-                ProviderRole.PRODUCER, ProviderRole.PROCESSOR,
-                ProviderRole.HOST
-            ],
-            url="https://github.com/stac-utils/stactools",
-        )
-    ]
+    providers = co.ALOS_PALSAR_PROVIDERS
 
     # Time must be in UTC
```
Comment (tracetechnical) [inline src/stactools/palsar/stac.py:37]: Is this  comment relevant if the demo_time has been removed?

File: src/stactools/palsar/commands.py [hunk 3]
```diff
@@ -21,35 +22,70 @@ def palsar():
         "create-collection",
         short_help="Creates a STAC collection",
     )
+    @click.argument("product")
     @click.argument("destination")
-    def create_collection_command(destination: str):
+    @click.option("-h",
```
Comment (vincentsarago) [inline src/stactools/palsar/commands.py:27]: `-h` should be reserved for `help`

File: src/stactools/palsar/cog.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        output_profile.update(dict(BIGTIFF="IF_SAFER"))
+
+        # Dataset Open option (see gdalwarp `-oo` option)
+        config = dict(
+            GDAL_NUM_THREADS="ALL_CPUS",
+            GDAL_TIFF_INTERNAL_MASK=True,
+            GDAL_TIFF_OVR_BLOCKSIZE="128",
+        )
+
+        if nodata is not None:
```
Comment (vincentsarago) [inline src/stactools/palsar/cog.py:65]: nodata will always be set so the if test is useless!

File: src/stactools/palsar/cog.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            cog_translate(
+                infile,
+                outfile,
+                output_profile,
+                config=config,
+                in_memory=False,
```
Comment (vincentsarago) [inline src/stactools/palsar/cog.py:71]: if `in_memory` is set to `None` (default), the application will decide if there is a need to use in_memory or in Disk temporary file https://github.com/cogeotiff/rio-cogeo/blob/master/rio_cogeo/cogeo.py#L247-L249

File: src/azure/requirements.txt [hunk 6]
```diff
@@ -5,5 +5,5 @@
 azure-functions
 azure-storage-blob
 azure-storage-queue
-rio-cogeo
+rio-cogeo==3.0.1
```
Comment (tracetechnical) [inline src/azure/requirements.txt:8]: I _think_ this can be dropped?

File: src/stactools/palsar/constants.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    AssetDefinition({
+        "title": "mask",
+        "type": "image/tiff; application=geotiff; profile=cloud-optimized",
+        "description": "Quality Mask",
+        "role": "data-mask"
+    }),
+}
+
+ALOS_FNF_ASSETS = {
+    "FNF":
```
Comment (TomAugspurger) [inline src/stactools/palsar/constants.py:95]: Should this be `"C"`, to match the band name below (https://github.com/developmentseed/stactools-palsar/pull/11/files#diff-dd520ef60a047d10abb3bc461035d029b8b0bc28378f447d83936ff6842cf273R120)? I haven't actually found where `asset_hrefs` is defined, which I think is what's determining this.

This is causing a failure in the tests for the Planetary Computer, which requires that the Collection's item_assets has the same keys as the assets on the items. But I might be working off an old item or collection.

**Pull request merged**